### PR TITLE
Setting up dev branch to be the v4 branch

### PR DESCRIPTION
**Pull request by vrdmr**
## Description
Migrating dev branch to work for v4 version of the Functions.

Fixes #

---

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/ci_e2e_workflow.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ 3.6, 3.7, 3.8, 3.9 ]
+        python-version: [ 3.6, 3.7, 3.8, 3.9, "3.10" ]
```
Comment (arroyc) [inline .github/workflows/ci_e2e_workflow.yml:24]: what does it mean when you enclose it with "" .. does it mean its ignore 3.10?

File: .flake8 [hunk 2]
```diff
@@ -8,6 +8,8 @@ ignore = W503,E402,E731
 exclude = .git, __pycache__, build, dist, .eggs, .github, .local, docs/,
           Samples, azure_functions_worker/protos/,
           azure_functions_worker/_thirdparty/typing_inspect.py,
-          tests/unittests/test_typing_inspect.py, .venv*, .env*, .vscode, venv*
+          tests/unittests/test_typing_inspect.py,
+          tests/unittests/broken_functions/syntax_error/main.py,
+          .venv*, .env*, .vscode, venv*, *.venv*,
```
Comment (arroyc) [inline .flake8:13]: is this assuming virtual environments will always be named like this?

File: .github/workflows/ci_e2e_workflow.yml (hunk 1 above)
Comment (vrdmr) [reply .github/workflows/ci_e2e_workflow.yml:24]: Ref: https://github.com/actions/setup-python/issues/249#issuecomment-934299359

**Comment by vrdmr** [comment]
Closing this in favor of #936.

**Pull request closed without merging**